src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -61,64 +68,95 @@ private LogParserPublisher(final boolean unstableOnWarning,
         this.projectRulePath = projectRulePath;
     }
 
+    @DataBoundConstructor
+    public LogParserPublisher(boolean useProjectRule, String projectRulePath, String parsingRulesPath) {
+        super();
+        if (useProjectRule) {
+            this.projectRulePath = Util.fixEmpty(projectRulePath);
+            this.parsingRulesPath = null;
+        } else {
+            this.parsingRulesPath = Util.fixEmpty(parsingRulesPath);
+            this.projectRulePath = null;
+        }
+        this.useProjectRule = useProjectRule;

```

Comment:
I've tried to modify the original source code as little as possible. `useProjectRule` can be the *flat data style*. It is represented by a radio button in the UI.